### PR TITLE
adds ability to support multi:softprob for xgboost

### PR DESCRIPTION
**Pull request by kgwinnup**
allows for use of multi:softprob for xgboost by defining a Softprob transformation and making the appropriate registration in transformation.go. Lastly, adds the transformation loading check for "multi:softprob" when loading an xgboost model.

### CODE REVIEW

File: transformation/softprob.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	sum := 0.0
+
+	for i, v := range rawPredictions {
+		exp := math.Exp(v)
+		outputPredictions[i] = exp
+		sum += exp
+	}
+
+	for i := range outputPredictions {
+		outputPredictions[i] /= sum
```
Comment (dmitryikh) [inline transformation/softprob.go:26]: softprob seems the same that softmax (with help of the SoftmaxFloat64Slice) already did?
Can you just reuse softmax transformation?

File: go.mod [hunk 2]
```diff
@@ -1,3 +1,3 @@
-module github.com/dmitryikh/leaves
+module github.com/kgwinnup/leaves
```
Comment (dmitryikh) [inline go.mod:1]: please use the path from this repo.

BTW, is it possible to omit github.. path  in every module?

Comment (kgwinnup) [reply go.mod:1]: Originally, I was using "replace" with gomod, but until/if this gets merged in the only solution I found was to just rename all the paths so I could continue with my current project. I don't know if you can omit the github part, possibly golang has some magic? Regardless, I will change this import as it definitely doesn't make sense in the PR.
